cargo-shuttle/src/args.rs

```diff
@@ -1,116 +1,122 @@
     Logs,
-    #[structopt(about = "delete the latest deployment for a shuttle project")]
+    #[clap(help = "delete the latest deployment for a shuttle project")]
     Delete,
-    #[structopt(about = "create user credentials for the shuttle platform")]
+    #[clap(help = "create user credentials for the shuttle platform")]
     Auth(AuthArgs),
-    #[structopt(about = "login to the shuttle platform")]
+    #[clap(help = "login to the shuttle platform")]
     Login(LoginArgs),
-    #[structopt(about = "run a shuttle project locally")]
+    #[clap(help = "run a shuttle project locally")]
     Run(RunArgs),
 }
 
-#[derive(StructOpt)]
+#[derive(Parser)]
 pub struct LoginArgs {
-    #[structopt(long, about = "api key for the shuttle platform")]
+    #[clap(long, help = "api key for the shuttle platform")]
     pub api_key: Option<String>,
 }
 
-#[derive(StructOpt)]
+#[derive(Parser)]
 pub struct AuthArgs {
-    #[structopt(about = "the desired username for the shuttle platform")]
+    #[clap(help = "the desired username for the shuttle platform")]
     pub username: String,
 }
 
-#[derive(StructOpt)]
+#[derive(Parser)]
 pub struct DeployArgs {
-    #[structopt(long, about = "allow dirty working directories to be packaged")]
+    #[clap(long, help = "allow dirty working directories to be packaged")]
     pub allow_dirty: bool,
-    #[structopt(long, about = "allows pre-deploy tests to be skipped")]
+    #[clap(long, help = "allows pre-deploy tests to be skipped")]
     pub no_test: bool,
 }
 
-#[derive(StructOpt, Debug)]
+#[derive(Parser, Debug)]
 pub struct RunArgs {
-    #[structopt(long, about = "port to start service on", default_value = "8000")]
+    #[clap(long, help = "port to start service on", default_value = "8000")]
     pub port: u16,
 }
 
-#[derive(StructOpt)]
+#[derive(Parser)]
 pub struct InitArgs {
-    #[structopt(
-        about = "the path to initialize a new shuttle project",
+    #[clap(
+        help = "the path to initialize a new shuttle project",
         parse(try_from_os_str = parse_init_path),
         default_value = ".",
     )]
     pub path: PathBuf,
 }
 
 // Helper function to parse and return the absolute path
-fn parse_path(path: &OsStr) -> Result<PathBuf, OsString> {
-    canonicalize(path).map_err(|e| format!("could not turn {path:?} into a real path: {e}").into())
+fn parse_path(path: &OsStr) -> Result<PathBuf, io::Error> {
+    canonicalize(path).map_err(|e| {
+        io::Error::new(
+            ErrorKind::InvalidInput,
+            format!("could not turn {path:?} into a real path: {e}"),
+        )
+    })
 }
 
 // Helper function to parse, create if not exists, and return the absolute path
-fn parse_init_path(path: &OsStr) -> Result<PathBuf, OsString> {
+fn parse_init_path(path: &OsStr) -> Result<PathBuf, io::Error> {
     // Create the directory if does not exist
-    create_dir_all(path).expect("could not find or create a directory with the given path");
+    create_dir_all(path)?;
```

Comment:
Instead of panicking, we can now just propagate this error.

Reply:
Nice :heart_eyes: